src/screens/EncounterScannerScreen.tsx

```diff
@@ -0,0 +1,137 @@
+import { device, texts } from '../config';
+import { useCreateEncounter, useEncounterUser } from '../hooks';
+import { User } from '../types';
+
+const showErrorAlert = () =>
+  Alert.alert(texts.encounter.errorScanTitle, texts.encounter.errorScanBody);
+
+const parseQrCode = (data: string): string | undefined => {
+  const result = Linking.parse(data);
+
+  if (result.scheme === (__DEV__ ? 'exp' : appJson.expo.scheme) && result.path === 'encounter') {
+    return result.queryParams?.qrId;
+  }
+};
+
+// TODO: accesibility labels
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const EncounterScannerScreen = ({ navigation }: { navigation: any }) => {
+  const { loading: loadingUser, userId } = useEncounterUser();
+  const [isScanning, setIsScanning] = useState(true);
+  const [hasPermission, setHasPermission] = useState<boolean>();
+
+  const onCreationSuccess = useCallback(
+    (data: User) => {
+      // TODO: add navigation here instead of console.log
+      // navigation.navigate(ScreenName.EncounterUserDetails, { data });
+      console.log(data);
+    },
+    [navigation]
+  );
+
+  const { loading: loadingCreateEncounter, createEncounter } = useCreateEncounter(
+    onCreationSuccess,
+    showErrorAlert
+  );
+
+  const handleBarCodeScanned = ({ data, type }: BarCodeScanningResult) => {
+    if (type !== (device.platform === 'android' ? 256 : 'org.iso.QRCode')) {
+      return;
+    }
+
+    setIsScanning(false);
+    const qrId = parseQrCode(data);
+    if (qrId) {
+      createEncounter(qrId, userId);
+    } else {
+      showErrorAlert();
+    }
+  };
+
+  const onPress = useCallback(() => {
+    setIsScanning(true);
+  }, []);
+
+  useEffect(() => {
+    (async () => {
+      const { status } = await Camera.requestCameraPermissionsAsync();
+      setHasPermission(status === 'granted');
+    })();
+  }, []);
+
+  if (hasPermission === undefined || loadingUser || loadingCreateEncounter) {
+    return (
+      <ScrollView>
+        <SectionHeader title={texts.encounter.scannerTitle} />
+        <LoadingSpinner loading />
+      </ScrollView>
+    );
+  }
+
+  if (!hasPermission) {
+    return (
+      <ScrollView>
+        <SectionHeader title={texts.encounter.scannerTitle} />
+        <WrapperWithOrientation>
+          <Wrapper>
+            <RegularText>{texts.encounter.cameraPermissionMissing}</RegularText>
+          </Wrapper>
+        </WrapperWithOrientation>
+      </ScrollView>
+    );
+  }
+
+  return (
+    <ScrollView>
+      <SectionHeader title={texts.encounter.scannerTitle} />
+      <WrapperWithOrientation>
+        <Wrapper>
+          <BoldText>{texts.encounter.scannerSubTitle}</BoldText>
+        </Wrapper>
+      </WrapperWithOrientation>
+      {isScanning ? (
+        <Wrapper style={styles.scannerContainer}>
+          <Camera
+            ratio="1:1"
+            onBarCodeScanned={handleBarCodeScanned}
+            style={styles.scanner}
+            // barCodeTypes={['org.iso.QRCode']}
```

Comment:
comment needed for later?

Reply:
nope, that is a comment that is a remainder of a previous version.
this prop is not supported here.

i removed it now